hasher-matcher-actioner/hmalib/models.py

```diff
@@ -48,6 +56,90 @@ def from_sns_message(cls, message: str) -> "SNSMessage":
         raise NotImplementedError
 
 
+@dataclass
+class SignalMetadataBase(DynamoDBItem):
+    """
+    Base for signal metadata.
+    'PG' refers to privacy group which for the time being is
+    quivalent to collab (and in the long term could map to bank)
+    """
+
+    PG_PREFIX = "pg#"
+
+    signal_id: t.Union[str, int]
+    pg_id: str
+    updated_at: datetime.datetime  # ISO-8601 formatted
+    signal_source: str
+    signal_hash: str  # duplicated field with PDQMatchRecord having both for now to help with debuging/testing
+    tags: t.List[t.Any] = field(default_factory=list)
+
+    @staticmethod
+    def get_dynamodb_pg_key(pg_id: str) -> str:
+        return f"{SignalMetadataBase.PG_PREFIX}{pg_id}"
+
+
+@dataclass
+class PDQSignalMetadata(SignalMetadataBase):
+    """
+    PDQ Signal metadata.
+    This object is designed to be an ~lookaside on some of the values used by
+    PDQMatchRecord for easier and more consistent updating by the syncer and UI.
+
+    Otherwise updates on a signals metadata would require updating all
+    PDQMatchRecord associated; TODO: For now there will be some overlap between
+    this object and PDQMatchRecord.
+    """
+
+    SIGNAL_TYPE = "pdq"
+
+    def to_dynamodb_item(self) -> dict:
+        return {
+            "PK": self.get_dynamodb_signal_key(self.signal_source, self.signal_id),
+            "SK": self.get_dynamodb_pg_key(self.pg_id),
+            "SignalHash": self.signal_hash,
+            "SignalSource": self.signal_source,
```

Comment:
blocking: Can you give examples of what SignalSource would be?

I'm concerned about the pg_key not being required to be globally unique.

What's preventing any other Source from having the same pg_id as another Source?

Reply:
They could, but I think that is okay because PK for this object incorporates the SignalSource as well. 'te' is what we have for now (i'll add a comment). It functions as a secondary prefix for PK atm (and SK for MatchRecord).